### PR TITLE
fozzie-components@2.23.0 - Storybook Webdriver refactor #globalconfig

### PR DESCRIPTION
**Pull request by siggerzz**
This PR includes the following:
- Removal of demo files
- Refactoring of CircleCI Config to build / serve static Storybook files
- Refactoring of testing npm scripts

Note: Some tests have been ignored as they will require refactoring, I how however verified locally that these tests run in Storybook and provide genuine test failures. In order to keep file changes to a minimum I will do the following in a separate PR:

- Refactor skipped tests
- Update README's

### CODE REVIEW

File: packages/f-button/package.json [hunk 1]
```diff
@@ -28,17 +28,23 @@
   "scripts": {
     "prepublishOnly": "yarn lint && yarn test && yarn build",
     "build": "vue-cli-service build --target lib --name f-button ./src/index.js",
-    "demo": "vue serve --port 8080 ./src/demo/Index.vue",
```
Comment (LTurns) [inline packages/f-button/package.json:31]: V small but I think the readme in f-button and f-alert might need updating

Comment (siggerzz) [reply packages/f-button/package.json:31]: README's will be updated as part of a separate PR. I just wanted to minimise the number of changed files here 👍

File: packages/f-footer/test-utils/component-objects/f-footer.component.js [hunk 2]
```diff
@@ -2,21 +2,23 @@ const footerComponent = () => $('[data-test-id="footer-component"]');
 const footerIcons = () => $$('[data-test-id="footerIcon"]');
 const iosIcon = () => footerIcons()[0];
 const androidIcon = () => footerIcons()[1];
-const facebookIcon = () => footerIcons()[2];
-const twitterIcon = () => footerIcons()[3];
-const instagramIcon = () => footerIcons()[4];
+const huaweiIcon = () => footerIcons()[1];
+const facebookIcon = () => footerIcons()[3];
+const twitterIcon = () => footerIcons()[4];
+const youtubeIcon = () => footerIcons()[5];
```
Comment (Preet-Sandhu) [inline packages/f-footer/test-utils/component-objects/f-footer.component.js:9]: Would these selectors break if they are presented in some other order? For e.g. if facebook icon is displayed at the end instead of this particular position.

Comment (siggerzz) [reply packages/f-footer/test-utils/component-objects/f-footer.component.js:9]: Yeah, I'll refactor this in another ticket to make it more robust 👍

File: packages/f-button/package.json (hunk 1 above)
Comment (LTurns) [reply packages/f-button/package.json:31]: cool, makes sense!

**Review by LTurns — DISMISSED** [review]

**Comment by Preet-Sandhu** [comment]
One of the original concerns was that Storybook takes a long time to launch as we try to initiate all the components? how does launching one single component compare. How much falser is this, and how does this compare with the previous approach?

**Comment by siggerzz** [comment]
> One of the original concerns was that Storybook takes a long time to launch as we try to initiate all the components? how does launching one single component compare. How much falser is this, and how does this compare with the previous approach?

Storybook now gets built / started before the component tests. All tests use this instance, so it's only started once, and takes roughly 50 seconds to start.

**Comment by Preet-Sandhu** [comment]
> > One of the original concerns was that Storybook takes a long time to launch as we try to initiate all the components? how does launching one single component compare. How much falser is this, and how does this compare with the previous approach?
> 
> Storybook now gets built / started before the component tests. All tests use this instance, so it's only started once, and takes roughly 50 seconds to start.

ok so as i understand it is being build with all the components rather than just a single one.  Would this be an issue as more and more components are added to the storybook in the future?

**Comment by siggerzz** [comment]
> > > One of the original concerns was that Storybook takes a long time to launch as we try to initiate all the components? how does launching one single component compare. How much falser is this, and how does this compare with the previous approach?
> > 
> > 
> > Storybook now gets built / started before the component tests. All tests use this instance, so it's only started once, and takes roughly 50 seconds to start.
> 
> ok so as i understand it is being build with all the components rather than just a single one. Would this be an issue as more and more components are added to the storybook in the future?

Nope, should be fine :)

File: packages/f-registration/test/specs/component/axe-accessibility.spec.js [hunk 3]
```diff
@@ -1,16 +1,16 @@
-import { getAccessibilityTestResults, processResults } from '../../../../../test/utils/axe-helper';
+// import { getAccessibilityTestResults, processResults } from '../../../../../test/utils/axe-helper';
```
Comment (siggerzz) [inline packages/f-registration/test/specs/component/axe-accessibility.spec.js:1]: This will be refactored in a separate PR

**Review by LTurns — APPROVED** [review]

File: packages/f-checkout/test/specs/component/f-checkout-collection.component.spec.js [hunk 4]
```diff
@@ -3,8 +3,14 @@ import DemoControls from '../../../test-utils/component-objects/demo-controls';
...
 describe('f-checkout "collection" component tests', () => {
+    beforeEach(() => {
+        browser.url('?path=/story/components-organisms--checkout-component');
+        browser.switchToFrame(0);
+        CheckoutComponent.waitForCheckoutComponent();
+    });
+
     forEach(['mobileNumber'])
-    .it('each fields error message should be displayed', field => {
+    .it.skip('each fields error message should be displayed', field => {
```
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout-collection.component.spec.js:13]: These will be refactored in a separate PR

File: packages/f-checkout/test/specs/component/f-checkout-delivery.component.spec.js [hunk 5]
```diff
@@ -2,8 +2,14 @@ import forEach from 'mocha-each';
...
 describe('f-checkout "delivery" component tests', () => {
+    beforeEach(() => {
+        browser.url('?path=/story/components-organisms--checkout-component');
+        browser.switchToFrame(0);
+        CheckoutComponent.waitForCheckoutComponent();
+    });
+
     forEach(['mobileNumber', 'addressLine1', 'addressCity', 'addressPostcode'])
-    .it('each fields error message should be displayed', field => {
+    .it.skip('each fields error message should be displayed', field => {
```
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout-delivery.component.spec.js:12]: These will be refactored in a separate PR

File: packages/f-checkout/test/specs/component/f-checkout.component.spec.js [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
 
 describe('f-checkout component tests', () => {
-    it('should display the f-checkout component', () => {
+    beforeEach(() => {
+        browser.url('?path=/story/components-organisms--checkout-component');
+        browser.switchToFrame(0);
+        CheckoutComponent.waitForCheckoutComponent();
+    });
+
+    it.skip('should display the f-checkout component', () => {
```
Comment (siggerzz) [inline packages/f-checkout/test/specs/component/f-checkout.component.spec.js:10]: These will be refactored in a separate PR

**Review by SachaZvetelman — APPROVED** [review]
Nice, @siggerzz.
We should update the `generator-component` as well in a separate PR.

**Comment by siggerzz** [comment]
@SachaZvetelman Already done! - https://github.com/justeat/fozzie-components/blob/815d6acc2f4441862854a6d908eb8a9489ba44a2/packages/generator-component/generators/app/templates/__package__.json#L37

**Comment by SachaZvetelman** [comment]
Missed that, @siggerzz! Thanks.
